File: EditImageView/src/main/java/com/image/edit/action/OnEditImageBaseActionListener.java
package com.image.edit.action;

import android.graphics.Canvas;
import android.support.annotation.NonNull;

import com.image.edit.EditImageView;
import com.image.edit.cache.EditImageCache;

/**
 * @author y
 * @create 2018/11/20
 */
public interface OnEditImageBaseActionListener {

    /**
     * 绘制
     *
     * @param editImageView {@link EditImageView}
     * @param canvas        {@link Canvas}
     */
    void onDraw(@NonNull EditImageView editImageView, @NonNull Canvas canvas);

    /**
     * 按下
     *
     * @param editImageView {@link EditImageView}
     * @param x             x
     * @param y             y
     */
    void onDown(@NonNull EditImageView editImageView, float x, float y);

    /**
     * 移动
     *
     * @param editImageView {@link EditImageView}
     * @param x             x
     * @param y             y
     */
    void onMove(@NonNull EditImageView editImageView, float x, float y);

    /**
     * 抬起
     *
     * @param editImageView {@link EditImageView}
     * @param x             x
     * @param y             y
     */
    void onUp(@NonNull EditImageView editImageView, float x, float y);

    /**
     * 保存缓存
     *
     * @param editImageView {@link EditImageView}
     */
    void onSaveImageCache(@NonNull EditImageView editImageView);

    /**
     * 回退到上一步
     *
     * @param editImageView  {@link EditImageView}
     * @param editImageCache {@link EditImageCache}
     */
    void onLastImageCache(@NonNull EditImageView editImageView, @NonNull EditImageCache editImageCache);
}

File: EditImageView/src/main/java/com/image/edit/action/OnEditImageBaseActionListener.kt
package com.image.edit.action

import android.graphics.Canvas
import com.image.edit.EditImageView
import com.image.edit.cache.EditImageCache

/**
 * @author y
 * @create 2018/11/20
 */

interface OnEditImageTextActionListener : OnEditImageBaseActionListener

interface OnEditImageEraserActionListener : OnEditImageBaseActionListener

interface OnEditImagePointActionListener : OnEditImageBaseActionListener

interface OnEditImageCustomActionListener : OnEditImageBaseActionListener

interface OnEditImageBaseActionListener {

    /**
     * 绘制
     *
     * @param editImageView [EditImageView]
     * @param canvas        [Canvas]
     */
    fun onDraw(editImageView: EditImageView, canvas: Canvas)

    /**
     * 按下
     *
     * @param editImageView [EditImageView]
     * @param x             x
     * @param y             y
     */
    fun onDown(editImageView: EditImageView, x: Float, y: Float)

    /**
     * 移动
     *
     * @param editImageView [EditImageView]
     * @param x             x
     * @param y             y
     */
    fun onMove(editImageView: EditImageView, x: Float, y: Float)

    /**
     * 抬起
     *
     * @param editImageView [EditImageView]
     * @param x             x
     * @param y             y
     */
    fun onUp(editImageView: EditImageView, x: Float, y: Float)

    /**
     * 保存缓存
     *
     * @param editImageView [EditImageView]
     */
    fun onSaveImageCache(editImageView: EditImageView)

    /**
     * 回退到上一步
     *
     * @param editImageView  [EditImageView]
     * @param editImageCache [EditImageCache]
     */
    fun onLastImageCache(editImageView: EditImageView, editImageCache: EditImageCache)
}
